modules/storages/lib/api/v3/file_links/create_endpoint.rb

```diff
@@ -0,0 +1,44 @@
+module API::V3::FileLinks
+  class CreateEndpoint < API::Utilities::Endpoints::Create
+    include ::API::V3::Utilities::Endpoints::V3Deductions
+    include ::API::V3::Utilities::Endpoints::V3PresentSingle
+
+    def process(request, params_elements)
+      global_result = ServiceResult.new(
+        success: true,
+        result: []
+      )
+      params_elements.each do |params|
+        one_result = super(request, params)
+        global_result.add_dependent!(one_result)
+        global_result.result << one_result.result
+      end
+      global_result
+    end
+
+    def present_success(request, call)
+      render_representer.create(
+        call.result,
+        self_link: request.api_v3_paths.file_links(request.work_package.id),
+        current_user: request.current_user
+      )
+    end
+
+    protected
+
+    def build_error_from_result(result)
+      ActiveModel::Errors.new result.first
```

Comment:
Do we expect the errors in all of the collection to be the same? Otherwise, we would have to introduce something like an `ErrorCollection`

Reply:
For now, the error handling of multiple file links creation is at the minimum. I'm not sure about the intended behavior here. What should happen when one file link creation fails to be created (invalid contract). Should we create all but the invalid one, or rollback everything?

My guess is that it's through early usage that we'll know which behavior is the best to adopt.

Reply:
Currently, If two file links are attempted to be created and one having an error, an error message is displayed but the other is created. That I think is weird. I'd opt for having it all within a transaction so that either all are created or non. But I would be ok with adding that in a separate PR.

Reply:
@cbliard If I remember right, @Kharonus's and my design idea was that posting a collection of `FileLink`s would create those `FileLink`s that were possible to create and will handle gracefully those that were not. So it would not be a "all or nothing" approach. The reasoning is the following: A file link is defined by very little data: a storage ID, container ID, and a file ID. The only things I see that could go wrong are:
- missing permission (should be the same check for all `FileLink`s as they get posted to the same container, i.e. work package) 
- incomplete/missing data (I assume that if that happens that usually all collection items are affected, i.e. host attribute)  
- violations of uniqueness constrains (which means the link was previously established. For the user all is good. Her request simply asked to have a state that is already present...a failed creation is not relevant for the user as the desired state is still achieved.... Posting a FileLink collection is kind of idempotent).

I don't say that this is the right way. For posting collections of other resources the desired behavior could be different. In our use case we thought

Reply:
I agree to @wielinde:
Our design decision was intended to not produce errors on file links, that won't be created. Usually this should happen due to the fk_constraint <storage_id, origin_file_id> to be already present. Hence, the POST will create only the links of the collection, that are not already there. It will fail only on missformatted request bodies, insufficient permissions or server errors. In addition, the response collection should contain only the really created ones, not the whole collection. At least, this is how the API is currently specced.

Reply:
Not sure, when a file link can have an error on creation? Aside of the "I'm already here"?

Reply:
At least, the error build here is usually one that occurs on contract validation. And this is permission mismatch or wrong storage data. Not sure about request format, if this is built here, too.

Reply:
incorrect/missing storage url, missing id, creation/update date not parseable (not iso8601) and so on. 

BTW, is there a limit in the id format? or can one store anything in it? I assume it is not only numeric.

Reply:
Our meeting discussion with Eric, Jens, Wieland, and myself has been captured in this example mapping: https://examplemapping.cucumber.io/examplemaps/cc0dec00-0e42-4f51-998d-6c12241e9068

It's about:
- what makes a file link payload valid / invalid
- how does the POST request handle valid / invalid / already existing / duplicate file link creation